client/pages/ReturnBike.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from "react";
+import { Text, View, StyleSheet, FlatList} from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+import BikeItem from '../components/BikeItem';
+import { rides } from '../data/testRides'; // switch to db/axios, use parameters
+import { bikes } from '../data/testBikes'; 
+
+const ReturnBike = ({navigation}) => {
+    const [uid, setUid] = useState('');
+    const [bike, setBike] = useState([]);
+    const [ride, setRide] = useState([]);
+
+    // get uid from local storage
+    const getUid = async () => {
+        const uid = await AsyncStorage.getItem('uid');
+        console.log(uid);
+        setUid(uid);
+    };
+
+    // grab current trip for the user
+    const getRide = async () => {
+        var result = rides.filter(function(val) {
+            return (val["rider"] == uid && val.end_time == null);
+        });
+        console.log(result);
+        setRide(result);
+    };
+
+    // grab bike associated with current trip
+    const getBike = async () => {
+        var result = bikes.filter(function(val) {
+            return (val["bike_id"] == ride[0].bike);
+        });
+        console.log(result);
+        setBike(result);
+    }
+
+    useEffect(() => {
+        getUid().then(uid => {
```

Comment:
I've been getting 'Possible Unhandled Promise Rejection' errors in my console and undefined variables due to this chain of calls. Also, in line 40 for example, shouldn't the _uid_ after the _getUid().then_ be the return value from _getUid()_? Maybe returning each successive data item down the line makes it eaier to manage?

Reply:
My intent was that the uid state variable would update first, then the ride state variable then the bike state variable. I've changed it so it returns values instead and passes them onto the next function as a param.

Reply:
Cool! I'm always getting stuck between state variables, return values, and so-called 'callback hell' so I'm not really sure which method is better.